app/src/metrics/README.md

```diff
@@ -0,0 +1,182 @@
+- `k`: Comma-separated list of k values (default: 5,10,25)
+- `questions_file`: Path to questions CSV file
+- `sampling`: Fraction of questions to sample (e.g., 0.1)
+- `min_score`: Minimum similarity score for retrieval (default: -1.0)
+- `commit`: Git commit hash of code being evaluated (auto-populated)
+
+### Log File Structure
+
+Each evaluation run creates four files in the logs directory:
+
+1. `batch_${UUID}.json` - Run metadata and configuration:
+```json
+{
+  "batch_id": "uuid",
+  "timestamp": "XXXZ",
+  "evaluation_config": {
+    "k_value": 5,
+    "dataset_filter": ["imagine_la"],
+    "num_samples": 100
+  },
+  "system_info": {
+    "package_version": "0.1.0",
+    "git_commit": "abc123"
+  }
+}
+```
+
+2. `results_${UUID}.jsonl` - Individual QA pair results:
+```json
+{
+  "qa_pair_id": "uuid",
+  "question": "What are the eligibility requirements?",
+  "expected_answer": "...",
+  "document_info": {
+    "name": "document.pdf",
+    "source": "imagine_la",
+    "chunk_id": "123",
+    "content_hash": "abc..."
+  },
+  "evaluation_result": {
+    "correct_chunk_retrieved": true,
+    "rank_if_found": 1,
+    "top_k_scores": [0.92, 0.85, 0.76],
+    "retrieval_time_ms": 150
+  },
+  "retrieved_chunks": [
+    {
+      "chunk_id": "123",
+      "score": 0.92,
+      "content": "..."
+    }
+  ]
+}
+```
+
+3. `results_${UUID}.csv` - Flattened results for analysis:
+- One row per retrieved chunk
+- Includes original result ID, chunk details, and retrieval metrics
+- Easier to analyze in spreadsheet tools
+
+4. `metrics_${UUID}.json` - Aggregated metrics:
+```json
+{
+  "batch_id": "uuid",
+  "timestamp": "XXXZ",
+  "overall_metrics": {
+    "recall_at_k": 0.90,
+    "mean_retrieval_time_ms": 145,
+    "total_questions": 100,
+    "successful_retrievals": 90,
+    "incorrect_retrievals_analysis": {
+      "incorrect_retrievals_count": 10,
+      "avg_score_incorrect": 0.45,
+      "datasets_with_incorrect_retrievals": ["dataset1", "dataset2"]
+    }
+  },
+  "dataset_metrics": {
+    "imagine_la": {
+      "recall_at_k": 0.90,
+      "sample_size": 100,
+      "avg_score_incorrect": 0.45
+    }
+  }
+}
+```
+
+### CSV Format
+
+The questions CSV file should have the following columns:
+- question: The question text
+- answer: Expected answer text
+- document_name: Name of source document
+- dataset: Dataset identifier (e.g., "Imagine LA")
+- chunk_id: ID of chunk containing answer
+- content_hash: Hash of chunk content for verification
+
+### Metrics Explanation
+
+The system computes several metrics to evaluate retrieval performance:
+
+- **Recall@k**: Whether the correct chunk was found within the top k results
+- **Mean Retrieval Time**: Average time in milliseconds to retrieve results
+- **Incorrect Retrievals Analysis**:
+  - Count of incorrect retrievals
+  - Average similarity score of incorrect retrievals
+  - Datasets where incorrect retrievals occurred (sorted by frequency)
```

Comment:
I suggest you move these descriptions inline when you present `metrics_${UUID}.json` above.